go/libkb/nist.go

```diff
@@ -77,10 +77,11 @@ func (t nistType) encoding() *base64.Encoding {
 type NISTFactory struct {
 	Contextified
 	sync.Mutex
-	uid      keybase1.UID
-	deviceID keybase1.DeviceID
-	key      GenericKey // cached secret signing key
-	nist     *NIST
+	uid                         keybase1.UID
+	deviceID                    keybase1.DeviceID
+	key                         GenericKey // cached secret signing key
+	nist                        *NIST
+	lastSuccessfulNISTShortHash []byte
```

Comment:
do we need to clear this on logout?

Reply:
why store the shorthash here rather than the NIST?

Reply:
>do we need to clear this on logout?
Should this be done at a higher level, clearing sensitive info from ActiveDevice (including nistFactory)?

>why store the shorthash here rather than the NIST?
Didn't want the entire thing to be kept around if there wasn't a reason for it, but I can change it if you think it's better

Reply:
>do we need to clear this on logout?

Should this be done at a higher level, clearing sensitive info from ActiveDevice (including nistFactory)?

>why store the shorthash here rather than the NIST?

Didn't want the entire thing to be kept around if there wasn't a reason for it, but I can change it if you think it's better

Reply:
test

Reply:
i think it feels slightly cleaner to keep the whole nist around...

also, the CI failure looks related, not a flake

Reply:
yeah, I reran after I merged server PR but it's still waiting on a CI executor.

Regards to clearing it, looks like `libkb.LogoutUsernameWithOptions -> GlobalContext.ActiveDevice.ClearGetKeychainMode() -> ActiveDevice.clear -> a.nistFactory = nil`, so it should be gone.